internal/compiler/generator/rust.rs

```diff
@@ -160,6 +160,36 @@ pub fn generate(
         return super::rust_live_preview::generate(doc, compiler_config);
```

Comment:
This should be an error if we are using libraries as this would AFAIK not work currently.
(we could try to make it work)